contracts/taxman/src/execute.rs

```diff
@@ -66,10 +71,15 @@ pub fn withhold_fee(ctx: SudoCtx, tx: Tx) -> StdResult<Response> {
         .add_attribute("withhold_amount", withhold_amount))
 }
 
-pub fn finalize_fee(ctx: SudoCtx, tx: Tx, outcome: TxOutcome) -> anyhow::Result<Response> {
+pub fn finalize_fee(ctx: AuthCtx, tx: Tx, outcome: TxOutcome) -> anyhow::Result<Response> {
     let cfg = CONFIG.load(ctx.storage)?;
     let info = ctx.querier.query_info()?;
 
+    // In simulation mode, don't do anything.
+    if ctx.mode == AuthMode::Simulate {
+        return Ok(Response::new());
+    }
+
```

Comment:
Why not set 
```rust   
    let fee_rate = if ctx.mode == AuthMode::Simulate {
        Dec128::Zero
   } else {
        config.fee_rate
   }
```
And using `fee_rate` instead of `config.fee_rate` on the next lines?